### DISCUSSION TITLE
Filament is completely ignoring my model casts

### DISCUSSION THREAD

**Discussion by yashdesai87** [filamentphp/filament#11191] [Help]
### Package

Table builder

### Package Version

3.0

### How can we help you?

I'm using the [spatie laravel model states](https://spatie.be/docs/laravel-model-states/v2/01-introduction) package to manage my application states and it is working great on the Laravel side of the application.

I'm using Filament for the admin backend. I have a column `listing_status` that stores an integer value of the status of a ticket. However, Filament displays my status value from the database as-is (integer) and completely ignores the casts put on the model.

My Ticket model below. The column `listing_status` in the database stores an integer value of the corresponding status.

```
class Tickets extends Model 
{
    use HasStates;

    protected $table = 'ticket';

    protected $casts = [
		'listing_status' => ListingStatus::class
	]
}
```

Below is the abstract class of the listing state:

    namespace App\States\ListingStatus;
    
    use Spatie\ModelStates\State;
    use Spatie\ModelStates\StateConfig;
    
    abstract class ListingStatus extends State
    {   
    	public static int $name;
    	public static string $label;
    }

Displayed on the panel using the TicketResource:

    public static function table(Table $table): Table
    {
        return $table
			->columns([
				TextColumn::make('listing_status')
			]);
	}

But all I see on the table are numeric values (photo attached below). Why is the cast being completely ignored? 

![enter image description here](https://i.ibb.co/JsrB9F3/Monosnap-Image-2024-02-01-00-11-34.png)

Curious as to what I may be doing wrong ?

**Comment by jeffreyvanhees** [comment]
Implement the `HasLabel`-interface on your base state class and add a `getLabel()` per state class. Then it will work out of the box! :)

```php
use Filament\Support\Contracts\HasLabel;

abstract class ListingStatus extends State implements HasLabel
{   
  public static string $name; // Not int, as in your example!
}
```

```php
class ListingStatusActive extends ListingStatus
{
  public static string $name = 'active';

  public function getLabel(): ?string
  {
    return 'Active';
  }
}
```

Also; there is now a package for spatie's model states to make it compatible with Filament!
https://filamentphp.com/plugins/maartenpaauw-model-states
